packages/contracts/tasks/helpers/Deployment.ts

```diff
@@ -354,6 +356,17 @@ export class Deployment {
     });
     await contract.deploymentTransaction()!.wait();
 
+    if (signer && (signer as AASigner).isAA) {
```

Comment:
This `isAA` attribute is read in the `deployContract` function. Just after the contract has been deployed, we check if the used signer is AA and if so we calculate the contract address from the transaction logs.

The returned contract if Signer isAA does not have the `.deploymentTransaction()` method

Reply:
won't the next couple of lines throw? 361 for instance

Reply:
No 😊 . I meant that the returned contract of this function does not has `.deploymentTransaction()` method because inside the function we are building the contract from a `BaseContract`